opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -223,7 +223,7 @@ namespace detail {
 
         // TODO: put this for now to avoid modify the following code.
         // TODO: this code can be fragile.
-        const Wells* wells_arg = &(asImpl().well_model_.wells());
+        const Wells* wells_arg = asImpl().well_model_.wellsPointer();
```

Comment:
This variable is not used except in the HAVE_MPI branch, could you move it inside while here? Would fix warnings when not using mpi, and be a little clearer.